### PR TITLE
Add show/screenshot to enable VPR GUI

### PR DESCRIPTION
**Pull request by azaidy**

### CODE REVIEW

File: siliconcompiler/tools/vpr/show.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    index = chip.get('arg', 'index')
+    tool, task = chip._get_tool_task(step, index)
+
+    options = vpr.runtime_options(chip)
+
+    if chip.valid('tool', tool, 'task', task, 'var', 'show_filepath'):
+        show_job = chip.get('tool', tool, 'task', task, 'var', 'show_job',
+                            step=step, index=index)
+        show_step = chip.get('tool', tool, 'task', task, 'var', 'show_step',
+                             step=step, index=index)
```
Comment (gadfort) [inline siliconcompiler/tools/vpr/show.py:47]: if this is false none of these variables will be set

File: siliconcompiler/tools/vpr/show.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        chip.error("Net file does not exist", fatal=True)
+
+    if route_file and place_file:
+        options.append('--analysis')
+        options.append(f'--place_file {place_file}')
+        options.append(f'--route_file {route_file}')
+    elif place_file:
+        options.append('--route')
+        options.append('--max_router_iterations 0')
+        options.append(f'--place_file {place_file}')
```
Comment (gadfort) [inline siliconcompiler/tools/vpr/show.py:75]: needs a comment as to why these options for route_itr

File: siliconcompiler/utils.py [hunk 3]
```diff
@@ -132,6 +134,8 @@ def get_default_iomap():
     default_iomap.update({ext: ('constraint', 'pcf') for ext in constraint_pcf})
 
     default_iomap.update({ext: ('fpga', 'xdc') for ext in fpga_xdc})
+    default_iomap.update({ext: ('layout', 'place') for ext in fpga_place})
+    default_iomap.update({ext: ('layout', 'route') for ext in fpga_route})
```
Comment (gadfort) [inline siliconcompiler/utils.py:138]: I think these should go into fpga/vpr_place and fpga/vpr_route to make it clear these are VPR files

File: siliconcompiler/tools/vpr/screenshot.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    design = chip.top()
+
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+    tool, task = chip._get_tool_task(step, index)
+
+    if chip.valid('tool', tool, 'task', task, 'var', 'show_filepath'):
+        show_type = chip.get('tool', tool, 'task', task, 'var', 'show_filetype',
+                             step=step, index=index)[0]
```
Comment (gadfort) [inline siliconcompiler/tools/vpr/screenshot.py:24]: if not valid this will error out

**Review by gadfort — CHANGES_REQUESTED** [review]
lets talk about this when you have time

File: siliconcompiler/targets/utils.py [hunk 5]
```diff
@@ -25,3 +29,10 @@ def set_common_showtools(chip):
     chip._load_module(openroad.__name__)
     chip._load_module(openroad_show.__name__)
     chip._load_module(openroad_screenshot.__name__)
+
+    # VPR
+    chip.set('option', 'showtool', 'place', 'vpr', clobber=False)
+    chip.set('option', 'showtool', 'route', 'vpr', clobber=False)
```
Comment (gadfort) [inline siliconcompiler/targets/utils.py:35]: switch the order of these two lines

File: siliconcompiler/tools/vpr/show.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                                     jobname=show_job[0])
+        net_file = chip.find_result('net', show_step[0],
+                                    jobname=show_job[0])
+        place_file = chip.find_result('place', show_step[0],
+                                      jobname=show_job[0])
+        route_file = chip.find_result('route', show_step[0],
+                                      jobname=show_job[0])
+
+    if blif_file:
+        options.append(f'{blif_file}')
```
[the remark this answers is not in the document]
Comment (azaidy) [reply siliconcompiler/tools/vpr/show.py]: Done

File: siliconcompiler/tools/vpr/show.py (hunk 1 above)
Comment (azaidy) [reply siliconcompiler/tools/vpr/show.py]: Done

File: siliconcompiler/tools/vpr/screenshot.py (hunk 4 above)
Comment (azaidy) [reply siliconcompiler/tools/vpr/screenshot.py:24]: Done

File: siliconcompiler/targets/utils.py (hunk 5 above)
Comment (azaidy) [reply siliconcompiler/targets/utils.py]: Done

File: siliconcompiler/tools/vpr/show.py (hunk 2 above)
Comment (azaidy) [reply siliconcompiler/tools/vpr/show.py:83]: Done

**Review by gadfort — APPROVED** [review]

**Pull request merged**